torch/_higher_order_ops/templated_attention.py

```diff
@@ -1,8 +1,9 @@
-from typing import Callable, Tuple
+from typing import Callable, Tuple, Union
 
 import torch
 import torch.utils._pytree as pytree
 from torch._C import DispatchKey
+from torch._dynamo.variables import UserFunctionVariable
```

Comment:
Don't do this, why does this HOP need to know about UserFunctionVariable?

Reply:
https://github.com/pytorch/pytorch/pull/123902/files#diff-d6c3a57794a9a8a79eacedc57d68ff631fe80776f8701398eb3ef349973b033bR1428

I left this comment on way I think flexattention does directly map to the existing patterns, I am very much a noob so if there is another way of doing this I am all ears

Reply:
Also these imports also cause some errors; https://github.com/pytorch/pytorch/pull/123902/files#diff-85e712896b6e0eb499cbfda5afd36d9fcf0c2988789cae78aeec8eb85c6fa18bR258-R267

Reply:
This was infact wrong